product_meat_unece/models/product_template.py

```diff
@@ -0,0 +1,170 @@
+    )
+    unece_meat_fat_thickness_id = fields.Many2one(
+        string="Fat thickness",
+        comodel_name="unece.code.list",
+        domain=[("type", "=", "meat_fat_thickness")],
+        ondelete="restrict",
+        unece_meat_code=True,
+    )
+    unece_meat_bovine_quality_system_id = fields.Many2one(
+        string="Bovine quality system",
+        comodel_name="unece.code.list",
+        domain=[("type", "=", "meat_bovine_quality_system")],
+        ondelete="restrict",
+        unece_meat_code=True,
+    )
+    unece_meat_weight_range_id = fields.Many2one(
+        string="Weight range",
+        comodel_name="unece.code.list",
+        domain=[("type", "=", "meat_weight_range")],
+        ondelete="restrict",
+        unece_meat_code=True,
+    )
+    unece_meat_packing_id = fields.Many2one(
+        string="Packing",
+        comodel_name="unece.code.list",
+        domain=[("type", "=", "meat_packing")],
+        ondelete="restrict",
+        unece_meat_code=True,
+    )
+    unece_meat_conformity_assessment_id = fields.Many2one(
+        string="Conformity assessment",
+        comodel_name="unece.code.list",
+        domain=[("type", "=", "meat_conformity_assessment")],
+        ondelete="restrict",
+        unece_meat_code=True,
+    )
+
+    @api.model
+    def _get_product_from_unece(self, product=False, **kwargs):
+        """
+        This method fetches the product template(s) matching the the given codes or
+        the given product template codes (if given).
+        Example: _get_product_from_unece(product_01, unece_meat_packing_id='1') will
+        search for a product template matching all the UNECE codes of product_01
+        but with unece_meat_packing_id.code == '1'.
+        :param product: product.template instance
+        :param kwargs: UNECE code values
+        :return: product.template instance != product
+        """
+        product_model = self.env["product.template"]
+        if not product:
+            product = product_model.browse()
+            domain = []
+        else:
+            domain = [("id", "!=", product.id)]
+        for product_field in self._fields.values():
+            if not product_field._attrs.get("unece_meat_code"):
+                continue
+            field_name = product_field.name
+            val = kwargs.get(field_name, False) or product[field_name].code
+            domain.append((field_name + ".code", "=", val))
+        return product_model.search(domain)
+
+    @api.depends(
+        "unece_meat_species_id",
+        "unece_meat_product_cut_id",
+        "unece_meat_refrigeration_id",
+        "unece_meat_bovine_category_id",
+        "unece_meat_production_system_id",
+        "unece_meat_feeding_system_id",
+        "unece_meat_post_slaughter_system_id",
+        "unece_meat_fat_thickness_id",
+        "unece_meat_bovine_quality_system_id",
+        "unece_meat_weight_range_id",
+        "unece_meat_packing_id",
+        "unece_meat_conformity_assessment_id",
```

Comment:
To have a perfect depends, you can use a lambda function into the `@api.depends()`, the function only have to return a `list` of fields (`str`).
A great improvement could be to return every fields with `unece_meat_code = True`.
So new modules with `unece_meat_code` will be automatically added.

Reply:
Good idea, done